org.lflang/src/org/lflang/generator/cpp/CppInstanceGenerator.kt

```diff
@@ -55,28 +49,29 @@ class CppInstanceGenerator(
             "std::unique_ptr<$cppType> $name;"
     }
 
-    private fun Instantiation.getParameterValue(param: Parameter, isBankInstantiation: Boolean = false): String? =
-        if (isBankInstantiation && param.name == "bank_index") {
-            // If we are in a bank instantiation (instanceId != null), then assign the instanceId
-            // to the parameter named "bank_index"
-            """__lf_idx"""
-        } else {
-            CppTypes.getCppArgumentForParameter(param, this)
+    private fun Instantiation.getParameterStruct(): String {
+        // If this is a bank instantiation and the instantiated reactor defines a "bank_index" parameter, we have to set
+        // bank_index here explicitly.
+        var prefix = "${cppType}::Parameters{"
+        if (isBank && reactor.hasBankIndexParameter())
+            prefix += ".bank_index = __lf_idx, "
+        return parameters.joinToString(", ", prefix, "}") {
+            val expr = it.rhs.exprs[0]
+            assert(!it.rhs.isBraces && !it.rhs.isParens && it.rhs.exprs.size == 0)
```

Comment:
I'm not sure this works... You're extracting an element at the line above so probably the assert should check for size == 1 (not 0) and be on the line above

Reply:
Oh, right. Also the assertion should come before the variable access. I think we never really use the weird syntax with multiple expressions, so the assertion war my way of checking.

I am seeing strange errors when I run all the tests (also in CI) that look like this:
```
Failed: src/concurrent/DeadlineHandledAboveThreaded.lf
    -----------------------------------------------------------------------------
    Reason: No information available. Exit code: ?
    Compilation output:
    Generating code for: file:/home/runner/work/lingua-franca/lingua-franca/test/Cpp/src/concurrent/DeadlineHandledAboveThreaded.lf
    ******** mode: STANDALONE
    ******** generated sources: /home/runner/work/lingua-franca/lingua-franca/test/Cpp/src-gen/concurrent/DeadlineHandledAboveThreaded
```
It dosen't provide a clear reason and I cannot reproduce the error when running lfc manually. So perhaps it is because the assertion is only executed during testing? But if so, then why is the error not properly reported in our tests?

Reply:
Assertions are only executed if the JVM is passed the flag `-ea`. Possibly the lfc launcher doesn't use this flag

Reply:
It doesn't..